src/runtime/runtime_program.c

```diff
@@ -29,13 +34,121 @@ void unload_program(Program* program) {
     free(program);
 }
 
-static bool extract_layout(SpecProgram * program) {
+static bool extract_entrypoint_info(const CompilerConfig* config, const Module* mod, EntryPointInfo* info) {
+    Nodes decls = get_module_declarations(mod);
+
+    const Node* args_struct_annotation;
+    const Node* args_struct_type = NULL;
+    const Node* entrypoint = NULL;
+
+    for (int i = 0; i < decls.count; ++i) {
+        const Node* node = decls.nodes[i];
+
+        switch (node->tag) {
+            case GlobalVariable_TAG:
+                const Node* entry_point_args_annotation = lookup_annotation(node, "EntryPointArgs");
```

Comment:
note: a statement must come after `case:` declarations are not statements 
https://en.cppreference.com/w/c/language/statements

(fixed in https://github.com/Hugobros3/shady/commit/e868a8aef02335b55f8333d036b9d24c60d3ae26)